pkg/actors/actors.go

```diff
@@ -1108,14 +1154,28 @@ func (a *actorsRuntime) CreateTimer(ctx context.Context, req *CreateTimerRequest
 					if err != nil {
 						log.Errorf("error invoking timer on actor %s: %s", actorKey, err)
 					}
-				} else {
+					if repeats > 0 {
+						cnt++
+					}
+				}
+				if !exists || (repeats > 0 && cnt >= repeats) {
 					a.DeleteTimer(ctx, &DeleteTimerRequest{
 						Name:      req.Name,
 						ActorID:   req.ActorID,
 						ActorType: req.ActorType,
 					})
 				}
+			case <-ttlTimerC:
+				a.DeleteTimer(ctx, &DeleteTimerRequest{
+					Name:      req.Name,
+					ActorID:   req.ActorID,
+					ActorType: req.ActorType,
+				})
 			case <-stop:
+				if ttl > 0 {
+					ttlTimer.Stop()
```

Comment:
When closing a timer, you need to check if the underlying channel needs to be drained.

https://pkg.go.dev/time#Timer.Stop